Adds a new option to core/editor to disable code editing (#14932)
WordPress/gutenberg: docs/designers-developers/developers/filters/editor-filters.md

wp.hooks.addFilter( 'editor.PostPreview.interstitialMarkup', 'my-plugin/custom-preview-message', customPreviewMessage );
```

## Editor settings

### `block_editor_settings`

> This is a PHP filter which is applied before sending settings to the Editor in `wp-admin/edit-form-blocks.php`.

Reviewer: ```suggestion
This is a PHP filter which is applied before sending settings to the WordPress block editor.
```

> You may find details about this filter [on its WordPress Code Reference page](wp-admin/edit-form-blocks.php).

Reviewer: This link is broken. It needs to be the full URL.

Author: ugh, bad copy paste apparently

The filter will send any setting to the initialized Editor, which means any editor setting that is used to configure the editor at initialisation can be filtered by a PHP WordPress plugin before being sent. 

### Available default editor settings

#### `richEditingEnabled`

> If it is true the user can edit the content using the visual editor.

Reviewer: ```suggestion
If it is `true` the user can edit the content using the Visual Editor.
```

> It is set by default by the WordPress function [`user_can_richedit`](https://developer.wordpress.org/reference/functions/user_can_richedit/). This function checks if the user can access the visual editor and that it’s supported by the user’s browser.

Reviewer: ```suggestion
It is set by default to the return value of the [`user_can_richedit`](https://developer.wordpress.org/reference/functions/user_can_richedit/) function. It checks if the user can access the Visual Editor and whether it’s supported by the user’s browser.
```

#### `codeEditingEnabled`

> Default `true`. Sets if the user can access the code editor **in adition** to the visual editor.

Reviewer: ```suggestion
Default `true`. Indicates whether the user can access the Code Editor **in addition** to the Visual Editor.
```

> If set to false the user will not be able to switch between Visual and Code editor. The option in the settings menu will not be available and the keyboard shortcut for switching editor types will not fire.  

Reviewer: ```suggestion
If set to `false` the user will not be able to switch to the Code Editor. The option in the settings menu will not be available and the keyboard shortcut for switching editor types will not fire.  
```
